lib/AST/Expr.cpp

```diff
@@ -126,7 +126,7 @@ namespace {
 } // end anonymous namespace
 
 void Expr::setType(Type T) {
-  assert(!T || !T->hasTypeVariable());
+  assert(!T || !T->hasTypeVariable() || !T->hasHole());
```

Comment:
What do you think about creating a `isCheckedType()` method that ensures that the type is not a `UNCHECKED_TYPE` node? That way we don't need to update every call site that does this: `!T->hasTypeVariable()`.

Reply:
That makes sense to me but I think we can't do it at the moment unless `UnresolvedType` is excluded from that check because code completion is still replying on solutions with unresolved types applied to AST (we are currently working on fixing that)...